apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/Fragment.kt

```diff
@@ -10,7 +10,7 @@ interface Fragment<D: Fragment.Data> {
 
   fun variables(): Operation.Variables
 
-  fun adapter(): ResponseAdapter<D>
+  fun adapter(responseAdapterCache: ResponseAdapterCache): ResponseAdapter<D>
```

Comment:
should it also take optional argument `responseAdapterCache: ResponseAdapterCache = ResponseAdapterCache.DEFAULT`

Reply:
Since this is mostly an internal API, I don't mind the mandatory parameter. I don't think many users will need it and those who need it definitely need to be aware of how what a `ResponseAdapterCache` is. 
Also this is bytecode that's going to end up in every Operation/Fragment. I usually like to put helpers method in extensions functions.